src/foundation.cs

```diff
@@ -8780,6 +8784,27 @@ partial interface NSItemProvider : NSCopying {
 #endif
 	}
 
+	interface INSItemProviderReading {}
+	
+	[Watch (4,0), TV (11,0), Mac (10,13), iOS (11,0)]
+	[Protocol]
+	interface NSItemProviderReading
+	{
+		[Static, Abstract]
+		[Export ("readableTypeIdentifiersForItemProvider", ArgumentSemantic.Copy)]
+		string[] ReadableTypeIdentifiersForItemProvider { get; }
+
+		//
+		// This is a constructor that various classes must implement
+		// NSAttributedString, UIColor and UIImage need to have this constructor
+		// for user-defined implementations of this interface, we are going to
+		// need to something special
+		//
+		//[Abstract]
+		//[Export ("initWithItemProviderData:typeIdentifier:error:")]
+		//INSItemProviderReading CreateFrom (NSData providerData, string typeIdentifier, [NullAllowed] NSError outError);
```

Comment:
As I've stated on the bug I believe the runtime doesn't need changes to make it work, so this can be uncommented (and the CreateFrom method should be a proper Constructor, not a method).

The registrars can provide some help for users, but it needs these methods properly defined in order to have enough information to help users.